src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -129,6 +157,91 @@ private Uri GetAccountUri(AzureBlobEgressProviderOptions options, out string acc
             return blobUriBuilder.ToUri();
         }
 
+        private Uri GetQueueAccountUri(AzureBlobEgressProviderOptions options, out string accountName)
+        {
+            var queueUriBuilder = new QueueUriBuilder(options.QueueAccountUri);
+
+            queueUriBuilder.Query = null;
+            queueUriBuilder.QueueName = null;
+
+            accountName = queueUriBuilder.AccountName;
+
+            return queueUriBuilder.ToUri();
+        }
+
+        public async Task EgressMessageToQueue(string artifactName, AzureBlobEgressProviderOptions options, CancellationToken token)
+        {
+            try
+            {
+                QueueClient queueClient = await GetQueueClientAsync(options, token);
+
+                if (queueClient.Exists())
+                {
+                    string queueMessage = ConstructQueueMessage(artifactName, options);
+                    queueClient.SendMessage(queueMessage);
+                }
+            }
+            catch (AggregateException ex) when (ex.InnerException is RequestFailedException innerException)
+            {
+                throw CreateException(innerException);
+            }
+            catch (RequestFailedException ex)
+            {
+                throw CreateException(ex);
+            }
+        }
+
+        private string ConstructQueueMessage(string artifactName, AzureBlobEgressProviderOptions options)
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream);
+
+            writer.WriteStartObject();
+
+            writer.WriteString("artifact", artifactName);
+            writer.WriteString("uri", options.AccountUri.ToString());
```

Comment:
May need to strip out the query portion before writing the account uri.

Reply:
Thinking about this some more, we should consider enabling something like https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-input?tabs=csharp#example. In this particular case, the payload wouldn't be json, but just the name of the file. 

This would greatly limit the amount of information about the blob that we would be sending via the queue, but it's likely that the person doing the egress would have a good understanding of the relationship between the queue and the blob already.

Reply:
I switched to just using the artifact name; if we change our minds I can always revert back to the previous format (or something else entirely).

Reply:
I think is is _okay_... I guess we could always add another option to have it write out a JSON object with more detailed information.

FYI, The [QueueTrigger] example that @wiktork linked to is not limited to just string values. It can, for the most part, use any deserializable type such as a POCO (which is implicitly understood to be a JSON object).